simulators/ethereum/consensus/testmodel.py

```diff
@@ -122,31 +143,41 @@ def get(self, key):
     def genesis(self, key = None):
         """ Returns the 'genesis' block for this testcase,
         including any alloc's (prestate) required """
+
+
         # Genesis block
         if self.raw_genesis is None:
+
+            #We must fix some fields in the genesis, geth is picky about that
+            fields_to_fix = ['nonce','coinbase','hash','mixHash','parentHash','receiptTrie','stateRoot','transactionsTrie','uncleHash']
+        
+
             raw_genesis = self.data['genesisBlockHeader']
 
-            # Turns out the testcases have noncewritten as 0102030405060708.
-            # Which is supposed to be interpreted as 0x0102030405060708.
-            # But if it's written as 0102030405060708 in the genesis file,
-            # it's interpreted differently. So we'll need to mod that on the fly
-            # for every testcase.
-            if 'nonce' in raw_genesis:
-                nonce = raw_genesis[u'nonce']
-                if not raw_genesis[u'nonce'][:2] == '0x':
-                    raw_genesis[u'nonce'] = '0x'+raw_genesis[u'nonce']
-
-            # Also, testcases have 'code' written as 0xdead
-            # But geth does not handle that, so we'll need to mod any of those also
-            # However, cpp-ethereum rejects 'code' written as 'dead'
-#            for addr, account in self.data['pre'].items():
-#                if 'code' in account:
-#                    code = account['code']
-#                    if code[:2] == '0x':
-#                        account['code'] = code[2:]
-
-
-            raw_genesis['alloc'] = self.data['pre']
+            for key in fields_to_fix:
+                v = raw_genesis[key]
+                if len(v) > 2 and v[:2] != '0x':
+                    raw_genesis[key] = '0x'+raw_genesis[key]
+
+            # And fix the alloc-section
+            alloc = {}
+
+            for addr in self.data['pre']:
+                v = self.data['pre'][addr]
+                if not addr[:2] == '0x':
+                    addr = '0x'+addr
```

Comment:
Ah, messy. Well, geth has become a lot more strict in this regard, so geth requires `0x`-prefixes in alloc. But actually.. `alloc` seems specified as `UnprefixedAddress` , https://github.com/ethereum/go-ethereum/blob/9fc90b6747c4eada258071cf65bebb6afd991a1c/core/genesis.go#L74 , which uses [`UnmarshalFixedUnprefixedText`](https://github.com/ethereum/go-ethereum/blob/8771c3061f340451d0966adcc547338a25f2231f/common/hexutil/json.go#L96) :

> decodes the input as a string with optional 0x prefix

I just tested a bit, and it appears to work fine doing the opposite:

            for addr in self.data['pre']:
                v = self.data['pre'][addr]
                if addr[:2] == '0x':
                    addr = addr[2:]